Adding CHS6540 (#350)
nanoframework/nanoFramework.IoT.Device: devices/Chs6540/README.md

# Chs6540 - Touch screen controller

> The Chs6540 is the touch screens controllers of the M5Stack [Tough](https://docs.m5stack.com/en/core/tough).

Reviewer: ```suggestion
The Chs6540 is the touch screen controller of the M5Stack [Tough](https://docs.m5stack.com/en/core/tough).
```

## Documentation

> - [More information](https://github.com/m5stack/M5Tough/blob/master/src/M5Touch.cpp) on the touch controller, unfortunately, there is datasheet available.

Reviewer: I believe there isn't a datasheet publicly available, correct? So this should should be

```suggestion
- [More information](https://github.com/m5stack/M5Tough/blob/master/src/M5Touch.cpp) on the touch controller, unfortunately, the datasheet isn't available.
```

Author: correct, there is none. So missed the not here :-D

## Usage

**Important**: make sure you properly setup the I2C pins especially for ESP32 before creating the `I2cDevice`, make sure you install the `nanoFramework.Hardware.ESP32 nuget`:

```csharp
//////////////////////////////////////////////////////////////////////
// when connecting to an ESP32 device, need to configure the I2C GPIOs
// used for the bus
Configuration.SetPinFunction(21, DeviceFunction.I2C1_DATA);
Configuration.SetPinFunction(22, DeviceFunction.I2C1_CLOCK);
```

For other devices like STM32, please make sure you're using the preset pins for the I2C bus you want to use.

> > Note: this sample requires a M5Core2.

Reviewer: ```suggestion
> Note: this sample requires a M5Stack Tought.
```

```csharp
using System.Device.Gpio;
using System.Device.I2c;
using System.Diagnostics;
using System.Threading;
using Iot.Device.Chs6540;
using nanoFramework.Hardware.Esp32;

//////////////////////////////////////////////////////////////////////
// when connecting to an ESP32 device, need to configure the I2C GPIOs
// used for the bus
Configuration.SetPinFunction(21, DeviceFunction.I2C1_DATA);
Configuration.SetPinFunction(22, DeviceFunction.I2C1_CLOCK);

I2cConnectionSettings settings = new(1, Chs6540.DefaultI2cAddress);
using I2cDevice device = I2cDevice.Create(settings);
using GpioController gpio = new();

using Chs6540 sensor = new(device);
sensor.SetInterruptMode(true);

gpio.OpenPin(39, PinMode.Input);
// This will enable an event on GPIO39 on falling edge when the screen if touched
gpio.RegisterCallbackForPinValueChangedEvent(39, PinEventTypes.Falling, TouchInterrupCallback);

while (true)
{
    Thread.Sleep(20);
}

void TouchInterrupCallback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
{
    Debug.WriteLine("Touch interrupt");
    var points = sensor.GetNumberPoints();
    if (points == 1)
    {
        var point = sensor.GetPoint(true);

>                 // Some controllers supports as well events, you can get access to them as well thru point.Event

Reviewer: ```suggestion
        // Some controllers supports as well events, you can get access to them as well thru point.Event
```
